app/templates/teachers/teacher-student-view.jade

```diff
@@ -186,44 +179,114 @@ block content
             if instance && instance.hasMember(view.user)
               - var levels = view.classroom.getLevels({courseID: course.id}).models
-              each level, index in levels
-                - var levelProgress = view.levelProgressMap[level.get('original')];
-                - courseProgress = courseProgress || levelProgress != 'not started';
-              if !_.isEmpty(view.levelStudentCodeMap) && courseProgress
-                table.table.table-condensed
-                  each level, index in levels
-                    if view.levelStudentCodeMap[level.get('original')]
-                      - var levelNumber = view.classroom.getLevelNumber(level.get('original'), index + 1);
-                      - var langClass = 'lang-python';
-                      if view.classroom.get('aceConfig')
-                        - langClass = 'lang-' + view.classroom.get('aceConfig').language;
-                      tr
-                        td(colspan=2)
-                          a(name=level.get('slug'))
-                          span Level #{levelNumber}: #{level.get('name')}
-                      tr
-                        td
-                          .small
-                            strong #{view.user.broadName()}
-                            span.spl(data-i18n="general.code")
-                            span :
+              - var courseStarted = false;
+              table.table.table-condensed
+                each level, index in levels
+                  if view.levelProgressMap[level.get('original')] === 'not started'
+                    - continue;
+                  - courseStarted = true;
+                  - var levelNumber = view.classroom.getLevelNumber(level.get('original'), index + 1);
+                  - var langClass = 'lang-python';
+                  if view.classroom.get('aceConfig')
+                    - langClass = 'lang-' + view.classroom.get('aceConfig').language;
+                  tr.code-level-row
+                    td(colspan=2)
+                      a(name=level.get('slug'))
+                      if level.get('assessment')
+                        span=levelNumber
+                        =" "
+                        span(data-i18n="play_level.level")
+                        span : #{level.get('name')}
+                      else
+                        span(data-i18n="play_level.level")
+                        =" "
+                        span #{levelNumber}: #{level.get('name')}
+                      if level.get('practice')
+                        =" ("
+                        span(data-i18n="teacher.practice")
+                        span )
+                      a.small.back-to-top(href='#backtotop', data-i18n="teacher.back_to_top")
+                  tr
+                    td
+                      .small
+                        span #{view.user.broadName()}'s
+                        span.spl(data-i18n="general.code")
```

Comment:
This `.small` isn't really translatable. It should be something like `"__name__'s code"` so it could be translated to something like `"le code de __name__"` 🇫🇷